testdata/redefines-builtin-id.go

```diff
@@ -32,3 +32,12 @@ var i, copy int // MATCH /redefinition of the built-in function copy/
 
 // issue #792
 type ()
+
+func foo() {
+	clear := 0 // MATCH /redefinition of the built-in function clear/
```

Comment:
revive itself requires go1.21. So I don't add any condition for go version here.
https://github.com/mgechev/revive/blob/master/go.mod#L3


I manually confirmed that the following code does not report any errors:

go.mod:
```go.mod
module playground

go 1.20
```
main.go
```go
// package comment
package main

func main() {
	clear := 0
	_ = clear
}
```

```
❯ revive -set_exit_status

❯ echo $?
0
```
```